### PR TITLE
T237 Facilitate specification of CORS-proxy

### PR DESCRIPTION
**Pull request by decentdav**
- Added specification of corsProxyURL() to src/utils.js
  - This is not used for imageUrls yet
- Fixed Error Toasts not showing the actual error message
- Fixed adding of new podcast subscriptions if not present on arlocal
  - src/client/index.js#feedDiff() did not take into account empty data structures
  - Reformatted GraphQL query
  - GraphQL querying now handles HTTP responses >= 400

### CODE REVIEW

File: src/client/arweave/get-podcast-feed.js [hunk 1]
```diff
@@ -18,27 +23,31 @@ export default async function getPodcastFeed(subscribeUrl) {
...
-          values: [subscribeUrl],
-        },
-      ],
-    },
-  })
-    .then(res => res.data.data.transactions.edges[0]?.node);
+    `
+  }
+
+  const edges = await client.api.post('/graphql', gqlQuery).then(res => {
```
Comment (mmahdigh) [inline src/client/arweave/get-podcast-feed.js:29]: You can ignore this comment, This is merely regarding our conversation that using `async/await` and `.then()` is [unconventional ](https://stackoverflow.com/questions/54385676/whats-wrong-with-awaiting-a-promise-chain/54387912#54387912 )and hard to reason about.

For instance, omitting `.then()`, line 29 to 35 can be written this way:

```
let edges;
try {
  const edgesRes = await client.api.post('/graphql', gqlQuery);
  edges = edgesRes.data.data.transactions.edges;
} catch (e) {
  edges = [];
}

```

File: src/client/rss/index.js [hunk 2]
```diff
@@ -7,27 +8,34 @@ function mergeItunesData(items, itunes) {
...
-        url: episode.enclosure?.url || episode.link || null,
-        publishedAt: publishedAt ? new Date(publishedAt) : null,
-        imageUrl: episode.image?.url || imageUrl,
-        categories: mergeItunesData(episode.categories, episode.itunes?.categories),
-        keywords: mergeItunesData(episode.keywords, episode.itunes?.keywords),
-      };
-    }),
-  };
+  try {
+    const { items, ...podcast } = await parser.parseURL(withCorsProxy(subscribeUrl));
```
Comment (mmahdigh) [inline src/client/rss/index.js:12]: In future, We can have some sort of validation in place to validate `subscribeUrl` to check if it's indeed a URL or not. Right now users can search any arbitrary term and surprisingly the RSS parser doesn't do any validation and carries on with making the request.

**Review by mmahdigh — CHANGES_REQUESTED** [review]

File: src/client/arweave/get-podcast-feed.js [hunk 3]
```diff
@@ -1,12 +1,17 @@
...
 import client from './client';
 import { toTag, fromTag } from './utils';
+import { isEmpty } from '../../utils';
 
 export default async function getPodcastFeed(subscribeUrl) {
-  const transaction = await client.api.post('/graphql', {
-    query: gql`
-      query GetPodcast($tags: [TagFilter!]!) {
-        transactions(tags: $tags, first: 100, sort: HEIGHT_DESC) {
+  const gqlTagFilter = `
```
[the remark this answers is not in the document]
Comment (decentdav) [reply src/client/arweave/get-podcast-feed.js]: It was quite a pain to get this working, but it's working now with GraphQL variables.

File: src/client/rss/index.js (hunk 2 above)
Comment (decentdav) [reply src/client/rss/index.js:12]: Added a TODO in `SubscriptionsProvider`.

**Review by decentdav — COMMENTED** [review]
Thanks for the sharp feedback!

File: src/client/arweave/get-podcast-feed.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (decentdav) [reply src/client/arweave/get-podcast-feed.js]: Back to the old version now. I think I made a mistake in my first patch where I thought the GraphQL query was not working.

File: src/client/arweave/get-podcast-feed.js [hunk 4]
```diff
@@ -22,23 +22,39 @@ export default async function getPodcastFeed(subscribeUrl) {
     variables: {
       tags: [
         {
-          name: toTag('subscribeUrl'),
-          values: [subscribeUrl],
+          name: `${toTag('subscribeUrl')}`,
```
Comment (decentdav) [inline src/client/arweave/get-podcast-feed.js:25]: This is redundant of course, will be fixed in the next patch, along with any new feedback.

**Comment by mmahdigh** [comment]
Looks good to me.

**Review by mmahdigh — APPROVED** [review]

**Pull request merged**